### ISSUE TITLE
bypassSecurityTrustStyle gives error on Safari

### ISSUE THREAD

**Issue by dolanmiu** [angular/angular#14910]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

Gives an error:
```
 inline template:1:28 caused by: Attempted to assign to readonly property.
```

**Expected behavior**

**Minimal reproduction of the problem with instructions**

Check out this Plunkr on **Chrome**, then check it out on **Safari**. Chrome has no error. **Safari has a bunch of errors**:

https://plnkr.co/edit/YETcOvKxhZsIbjpd3FLh?p=preview

Here is a plunkr with no error on both (I removed bypassSecurityTrustStyle in the template):

https://plnkr.co/edit/BCpyitIqIGCc44k1w9Q9?p=preview

**What is the motivation / use case for changing the behavior?**

Because its good to have it working on both browsers

**Please tell us about your environment:**

Mac OS Latest, fully updated. 

* **Angular version:** 2.0.X

Angular v2.4.9

* **Browser:** Safari 10.0.3

 
* **Language:** [TypeScript 2]

**Comment by DzmitryShylovich** [comment]
can you try with angular 4.x?

**Comment by dolanmiu** [comment]
Isnt the latest angular 2.4.9? Angular4 isnt released?

**Comment by DzmitryShylovich** [comment]
angular 4 rc2 is released.
here's the plunkr http://plnkr.co/edit/lU32vBJY9KYXZUTW7GQQ?p=preview

**Comment by DzmitryShylovich** [comment]
> Can you see the issue your end though through the Plunkrs?

nope, I'm on windows :)

**Comment by dolanmiu** [comment]
http://plnkr.co/edit/EVTnT2vGIwsZU18pn5Fx?p=preview

here is the re-created updated plunkr, based on the plunkr you gave

same error!

**Comment by dolanmiu** [comment]
![screen shot 2017-03-03 at 17 37 27](https://cloud.githubusercontent.com/assets/2917613/23561886/20fed152-0038-11e7-93d1-6e94a3c78c1e.png)

Here is a screenshot of it

**Comment by wKoza** [comment]
It doesn't seems to be a problem with bypassSecurityTrustStyle but with Safari, [style] doesn't work. If you specify the css property, like  [style.left] , it will work. 

For instance : https://plnkr.co/edit/we79NR9vu1YpntAEniGc?p=preview

@DzmitryShylovich if it can help you.

**Comment by Toxicable** [comment]
Looks to be an issue with Safari and not Angular in this case then

**Comment by wKoza** [comment]
Angular must be platform agnostic for me.

**Comment by dolanmiu** [comment]
@wKoza yes I agree, it must work the same in all platforms regardless. The developer shouldn't need to put safari specific hacks in.

P.S. this bug also happens in Ionic 2 on iOS, cos I am guessing iPhone internally uses Safari engine for hybrid apps

**Comment by DzmitryShylovich** [comment]
it's not a "bug". this is how Safari works in a strict mode.

**Comment by dolanmiu** [comment]
But in that plunkr; how does one turn off strict mode?

**Comment by wKoza** [comment]
Why don't you use [ngStyle] rather ? https://plnkr.co/edit/nZF1QFvKaAxfRgAofIKR?p=preview

**Comment by connor4312** [comment]
@DzmitryShylovich I can consistently reproduce this in 4.1

**Comment by JeanMeche** [comment]
Closing as not relevant anymore.
